### PR TITLE
Neutron DHCP Agent adoption

### PR DESCRIPTION
**Pull request by umago**
Adds adoption of the neutron-dhcp-agent for edpm nodes.

This patch also changes documentation and add a simple test to ensure that the neutron-dhcp-agent is ALIVE.

Depends-On: https://github.com/openstack-k8s-operators/install_yamls/pull/767

Closes: OSPRH-2435

### CODE REVIEW

File: docs_user/modules/openstack-edpm_adoption.adoc [hunk 1]
```diff
@@ -406,6 +406,18 @@ oc patch openstackdataplanenodeset openstack --type='json' --patch='[
...
 
+* Optionally enable neutron-dhcp-agent in the OpenStackDataPlaneNodeSet CR
++
+[source,yaml]
+----
+oc patch openstackdataplanenodeset openstack --type='json' --patch='[
+  {
+    "op": "add",
+    "path": "/spec/services/-",
+    "value": "neutron-dhcp-agent"
```
Comment (slawqo) [inline docs_user/modules/openstack-edpm_adoption.adoc:417]: service is called "neutron-dhcp"

Comment (umago) [reply docs_user/modules/openstack-edpm_adoption.adoc:417]: Thanks! Lemme fix this

**Review by jistr — APPROVED** [review]

**Comment by umago** [comment]
recheck

**Review by slawqo — APPROVED** [review]
/lgtm

**Comment by umago** [comment]
recheck

**Comment by slawqo** [comment]
recheck

**Comment by umago** [comment]
recheck

**Comment by booxter** [comment]
From log

```
2024-03-25 14:46:09.570720 | controller | fatal: [localhost]: FAILED! => {"changed": true, "msg": "non-zero return code", "rc": 255, "stderr": "", "stderr_lines": [], "stdout": "Executing: program /usr/bin/ssh host api.crc.testing, user root, command sftp\nOpenSSH_8.7p1, OpenSSL 3.0.7 1 Nov 2022\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config.d/50-redhat.conf\r\ndebug1: Reading configuration data /etc/crypto-policies/back-ends/openssh.config\r\ndebug1: configuration requests final Match pass\r\ndebug1: re-parsing configuration\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config.d/50-redhat.conf\r\ndebug1: Reading configuration data /etc/crypto-policies/back-ends/openssh.config\r\ndebug1: Connecting to api.crc.testing [38.102.83.235] port 22.\r\ndebug1: Connection established.\r\ndebug1: identity file /home/zuul/.ssh/id_rsa type 0\r\ndebug1: identity file /home/zuul/.ssh/id_rsa-cert type -1\r\ndebug1: Local version string SSH-2.0-OpenSSH_8.7\r\ndebug1: Remote protocol version 2.0, remote software version OpenSSH_8.7\r\ndebug1: compat_banner: match: OpenSSH_8.7 pat OpenSSH* compat 0x04000000\r\ndebug1: Authenticating to api.crc.testing:22 as 'root'\r\ndebug1: load_hostkeys: fopen /home/zuul/.ssh/known_hosts2: No such file or directory\r\ndebug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory\r\ndebug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory\r\ndebug1: SSH2_MSG_KEXINIT sent\r\ndebug1: SSH2_MSG_KEXINIT received\r\ndebug1: kex: algorithm: curve25519-sha256\r\ndebug1: kex: host key algorithm: ssh-ed25519\r\ndebug1: kex: server->client cipher: aes256-gcm@openssh.com MAC: <implicit> compression: none\r\ndebug1: kex: client->server cipher: aes256-gcm@openssh.com MAC: <implicit> compression: none\r\ndebug1: kex: curve25519-sha256 need=32 dh_need=32\r\ndebug1: kex: curve25519-sha256 need=32 dh_need=32\r\ndebug1: expecting SSH2_MSG_KEX_ECDH_REPLY\r\ndebug1: SSH2_MSG_KEX_ECDH_REPLY received\r\ndebug1: Server host key: ssh-ed25519 SHA256:n4aJR8/mtezVJWmHCxv4Q4WAcPOE426fbrOpQZPMzo8\r\ndebug1: load_hostkeys: fopen /home/zuul/.ssh/known_hosts2: No such file or directory\r\ndebug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory\r\ndebug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory\r\ndebug1: Host 'api.crc.testing' is known and matches the ED25519 host key.\r\ndebug1: Found key in /home/zuul/.ssh/known_hosts:10\r\ndebug1: rekey out after 4294967296 blocks\r\ndebug1: SSH2_MSG_NEWKEYS sent\r\ndebug1: expecting SSH2_MSG_NEWKEYS\r\ndebug1: SSH2_MSG_NEWKEYS received\r\ndebug1: rekey in after 4294967296 blocks\r\ndebug1: Will attempt key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit\r\ndebug1: SSH2_MSG_EXT_INFO received\r\ndebug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,sk-ssh-ed25519@openssh.com,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ecdsa-sha2-nistp256@openssh.com,webauthn-sk-ecdsa-sha2-nistp256@openssh.com>\r\ndebug1: SSH2_MSG_SERVICE_ACCEPT received\r\ndebug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic\r\ndebug1: Next authentication method: gssapi-with-mic\r\ndebug1: No credentials were supplied, or the credentials were unavailable or inaccessible\nNo Kerberos credentials available (default cache: KCM:)\n\n\r\ndebug1: No credentials were supplied, or the credentials were unavailable or inaccessible\nNo Kerberos credentials available (default cache: KCM:)\n\n\r\ndebug1: Next authentication method: publickey\r\ndebug1: Offering public key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit\r\ndebug1: Server accepts key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit\r\ndebug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic\r\ndebug1: No more authentication methods to try.\r\nroot@api.crc.testing: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).\r\nConnection closed\r\n", "stdout_lines": ["Executing: program /usr/bin/ssh host api.crc.testing, user root, command sftp", "OpenSSH_8.7p1, OpenSSL 3.0.7 1 Nov 2022", "debug1: Reading configuration data /etc/ssh/ssh_config", "debug1: Reading configuration data /etc/ssh/ssh_config.d/50-redhat.conf", "debug1: Reading configuration data /etc/crypto-policies/back-ends/openssh.config", "debug1: configuration requests final Match pass", "debug1: re-parsing configuration", "debug1: Reading configuration data /etc/ssh/ssh_config", "debug1: Reading configuration data /etc/ssh/ssh_config.d/50-redhat.conf", "debug1: Reading configuration data /etc/crypto-policies/back-ends/openssh.config", "debug1: Connecting to api.crc.testing [38.102.83.235] port 22.", "debug1: Connection established.", "debug1: identity file /home/zuul/.ssh/id_rsa type 0", "debug1: identity file /home/zuul/.ssh/id_rsa-cert type -1", "debug1: Local version string SSH-2.0-OpenSSH_8.7", "debug1: Remote protocol version 2.0, remote software version OpenSSH_8.7", "debug1: compat_banner: match: OpenSSH_8.7 pat OpenSSH* compat 0x04000000", "debug1: Authenticating to api.crc.testing:22 as 'root'", "debug1: load_hostkeys: fopen /home/zuul/.ssh/known_hosts2: No such file or directory", "debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory", "debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory", "debug1: SSH2_MSG_KEXINIT sent", "debug1: SSH2_MSG_KEXINIT received", "debug1: kex: algorithm: curve25519-sha256", "debug1: kex: host key algorithm: ssh-ed25519", "debug1: kex: server->client cipher: aes256-gcm@openssh.com MAC: <implicit> compression: none", "debug1: kex: client->server cipher: aes256-gcm@openssh.com MAC: <implicit> compression: none", "debug1: kex: curve25519-sha256 need=32 dh_need=32", "debug1: kex: curve25519-sha256 need=32 dh_need=32", "debug1: expecting SSH2_MSG_KEX_ECDH_REPLY", "debug1: SSH2_MSG_KEX_ECDH_REPLY received", "debug1: Server host key: ssh-ed25519 SHA256:n4aJR8/mtezVJWmHCxv4Q4WAcPOE426fbrOpQZPMzo8", "debug1: load_hostkeys: fopen /home/zuul/.ssh/known_hosts2: No such file or directory", "debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory", "debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory", "debug1: Host 'api.crc.testing' is known and matches the ED25519 host key.", "debug1: Found key in /home/zuul/.ssh/known_hosts:10", "debug1: rekey out after 4294967296 blocks", "debug1: SSH2_MSG_NEWKEYS sent", "debug1: expecting SSH2_MSG_NEWKEYS", "debug1: SSH2_MSG_NEWKEYS received", "debug1: rekey in after 4294967296 blocks", "debug1: Will attempt key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit", "debug1: SSH2_MSG_EXT_INFO received", "debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,sk-ssh-ed25519@openssh.com,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ecdsa-sha2-nistp256@openssh.com,webauthn-sk-ecdsa-sha2-nistp256@openssh.com>", "debug1: SSH2_MSG_SERVICE_ACCEPT received", "debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic", "debug1: Next authentication method: gssapi-with-mic", "debug1: No credentials were supplied, or the credentials were unavailable or inaccessible", "No Kerberos credentials available (default cache: KCM:)", "", "", "debug1: No credentials were supplied, or the credentials were unavailable or inaccessible", "No Kerberos credentials available (default cache: KCM:)", "", "", "debug1: Next authentication method: publickey", "debug1: Offering public key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit", "debug1: Server accepts key: /home/zuul/.ssh/id_rsa RSA SHA256:f7i/tx1SCFSGvPFumUleJ+k3vX15zHszvrSkeWe1pwk explicit", "debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic", "debug1: No more authentication methods to try.", "root@api.crc.testing: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).", "Connection closed"]}
```

```
2024-03-25 14:46:10.160749 | controller | fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'SSH_KEY_FILE'. 'dict object' has no attribute 'SSH_KEY_FILE'\n\nThe error appears to be in '/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles/artifacts/tasks/edpm.yml': line 39, column 11, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n        - name: Extract Compute from zuul mapping if any\n          ^ here\n"}
```

**Comment by umago** [comment]
Apart from the error @booxter has posted which seems related to ci-framework where those files are present, the adoption CI is also timing out at an early stage with the following error:

`2024-03-26 06:11:21.265028 | controller | TASK [ci_local_storage : Perform operation on worker node _raw_params=oc debug node/{{ item }} -T -- chroot /host /usr/bin/bash -c "for pvnum in $(seq -w -s ' ' 01 {{ cifmw_cls_pv_count }}); do echo \"{{ _msg }} dir {{ cifmw_cls_local_storage_name }}/pv\${pvnum} on {{ item }}\"; {{ cifmw_cls_action }} {{ cifmw_cls_local_storage_name }}/pv\${pvnum}; done"] ***
2024-03-26 06:11:21.265033 | controller | Tuesday 26 March 2024  06:11:21 -0400 (0:00:00.055)       0:00:09.628 *********
2024-03-26 12:57:50.308164 | PRE-RUN END RESULT_TIMED_OUT: [untrusted : review.rdoproject.org/rdo-jobs/playbooks/data_plane_adoption/setup_crc_use_install_yamls_extracted_crc.yaml@maste`

I looked at other PRs proposed for this project and it's happening there too. Nothing has changed in this task for the last 2 months [0]. Trying to figure out what else has changed to be causing this.

[0] https://github.com/openstack-k8s-operators/ci-framework/blame/370b0ff4d62e99e5794a6438950c7f5249a0a127/roles/ci_local_storage/tasks/worker_node_dirs.yml#L9

**Comment by umago** [comment]
recheck

I believe [0] has fixed the problem mentioned before.

[0] https://github.com/openstack-k8s-operators/ci-framework/commit/e52b37a02f8674f2d55701abc21b2b62e7e64942

**Comment by umago** [comment]
recheck

**Comment by umago** [comment]
The "data-plane-adoption-osp-17-to-extracted-crc" job seems stable now.

The "data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph" job is failing at the "dataplane_adoption : wait for dataplane node set to be ready" task [0] and does not look related to the changes.

[0] https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-03-28T11:01:23EDT.log

**Comment by karelyatin** [comment]
> The "data-plane-adoption-osp-17-to-extracted-crc" job seems stable now.
> 
> The "data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph" job is failing at the "dataplane_adoption : wait for dataplane node set to be ready" task [0] and does not look related to the changes.
> 
> [0] https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-03-28T11:01:23EDT.log

The task not finished in 30 minutes timeout currently set, it took around 32 minute to complete.
I see there is a draft patch already to increase that timeout:- https://github.com/openstack-k8s-operators/data-plane-adoption/pull/390

Also seems some issue with dhcp agent startup as seeing some permission issues with log dir, so those needs to be checked too:-  https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/standalone/containers/stdouts/neutron_dhcp.log

**Comment by umago** [comment]
recheck

**Comment by karelyatin** [comment]
recheck RETRY_LIMIT reboot timeout

**Comment by slawqo** [comment]
recheck

**Comment by umago** [comment]
recheck

**Comment by karelyatin** [comment]
> > > The "data-plane-adoption-osp-17-to-extracted-crc" job seems stable now.
> > > The "data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph" job is failing at the "dataplane_adoption : wait for dataplane node set to be ready" task [0] and does not look related to the changes.
> > > [0] https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-03-28T11:01:23EDT.log
> > 
> > 
> > The task not finished in 30 minutes timeout currently set, it took around 32 minute to complete. I see there is a draft patch already to increase that timeout:- #390
> 
> Thanks Yatin for the pointer!
> 
> > Also seems some issue with dhcp agent startup as seeing some permission issues with log dir, so those needs to be checked too:- https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/standalone/containers/stdouts/neutron_dhcp.log
> 
> I will take a look.

I see same errors for other agents too. With this seems there running multiple dhcp and other agents(one original and one after adoption)? don't we need to stop the original one? Not enough logs collected in CI to confirm this, wondering if you tested it locally and observed something like this?

**Comment by karelyatin** [comment]
> > > > The "data-plane-adoption-osp-17-to-extracted-crc" job seems stable now.
> > > > The "data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph" job is failing at the "dataplane_adoption : wait for dataplane node set to be ready" task [0] and does not look related to the changes.
> > > > [0] https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-03-28T11:01:23EDT.log
> > > 
> > > 
> > > The task not finished in 30 minutes timeout currently set, it took around 32 minute to complete. I see there is a draft patch already to increase that timeout:- #390
> > 
> > 
> > Thanks Yatin for the pointer!
> > > Also seems some issue with dhcp agent startup as seeing some permission issues with log dir, so those needs to be checked too:- https://logserver.rdoproject.org/49/349/69854b68d84edf7d6557bb435c0a308d3e1f81d5/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/d4d7f7f/standalone/containers/stdouts/neutron_dhcp.log
> > 
> > 
> > I will take a look.
> 
> I see same errors for other agents too. With this seems there running multiple dhcp and other agents(one original and one after adoption)? don't we need to stop the original one? Not enough logs collected in CI to confirm this, wondering if you tested it locally and observed something like this?

Ok found sos reports collected:-
https://logserver.rdoproject.org/49/349/7cb8b39e2b563d7c724a56b64d4a57c111ea621a/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/f9b09be/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/sos-reports/_all_nodes/sosreport-standalone.localdomain/sos_commands/systemd/systemctl_status_--all

â— tripleo_neutron_dhcp.service - neutron_dhcp container
     Loaded: loaded (/etc/systemd/system/tripleo_neutron_dhcp.service; enabled; preset: disabled)
     Active: active (running) since Wed 2024-04-03 23:00:54 UTC; 49min ago
   Main PID: 97413 (conmon)
      Tasks: 0 (limit: 100220)
     Memory: 16.0K
        CPU: 287ms
     CGroup: /system.slice/tripleo_neutron_dhcp.service
             â€£ 97413 /usr/bin/conmon --api-version 1 -c e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c -u e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c -r /usr/bin/crun -b /var/lib/containers/storage/overlay-containers/e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c/userdata -p /run/containers/storage/overlay-containers/e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c/userdata/pidfile -n neutron_dhcp --exit-dir /run/libpod/exits --full-attach -s -l k8s-file:/var/log/containers/stdouts/neutron_dhcp.log --log-level warning --syslog --runtime-arg --log-format=json --runtime-arg --log --runtime-arg=/run/containers/storage/overlay-containers/e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c/userdata/oci-log --conmon-pidfile /run/neutron_dhcp.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /var/lib/containers/storage --exit-command-arg --runroot --exit-command-arg /run/containers/storage --exit-command-arg --log-level --exit-command-arg warning --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/libpod --exit-command-arg --network-config-dir --exit-command-arg "" --exit-command-arg --network-backend --exit-command-arg netavark --exit-command-arg --volumepath --exit-command-arg /var/lib/containers/storage/volumes --exit-command-arg --db-backend --exit-command-arg boltdb --exit-command-arg --transient-store=false --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg --storage-opt --exit-command-arg overlay.mountopt=nodev,metacopy=on --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg container --exit-command-arg cleanup --exit-command-arg e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c

Apr 03 23:00:53 standalone.localdomain systemd[1]: Starting neutron_dhcp container...
Apr 03 23:00:53 standalone.localdomain tripleo-start-podman-container[101298]: Creating additional drop-in dependency for "neutron_dhcp" (e9de9cf6d3b0f81680e0a273d1f1dadbf9dc9f06d304c091e7173803b1b48b2c)
Apr 03 23:00:54 standalone.localdomain systemd[1]: Started neutron_dhcp container.


â— edpm_neutron_dhcp_agent.service - neutron_dhcp_agent container
     Loaded: loaded (/etc/systemd/system/edpm_neutron_dhcp_agent.service; enabled; preset: disabled)
     Active: active (running) since Wed 2024-04-03 23:44:23 UTC; 6min ago
   Main PID: 317396 (conmon)
      Tasks: 1 (limit: 100220)
     Memory: 652.0K
        CPU: 156ms
     CGroup: /system.slice/edpm_neutron_dhcp_agent.service
             â””â”€317396 /usr/bin/conmon --api-version 1 -c 695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c -u 695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c -r /usr/bin/crun -b /var/lib/containers/storage/overlay-containers/695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c/userdata -p /run/containers/storage/overlay-containers/695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c/userdata/pidfile -n neutron_dhcp_agent --exit-dir /run/libpod/exits --full-attach -s -l journald --log-level warning --syslog --runtime-arg --log-format=json --runtime-arg --log --runtime-arg=/run/containers/storage/overlay-containers/695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c/userdata/oci-log --conmon-pidfile /run/neutron_dhcp_agent.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /var/lib/containers/storage --exit-command-arg --runroot --exit-command-arg /run/containers/storage --exit-command-arg --log-level --exit-command-arg warning --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/libpod --exit-command-arg --network-config-dir --exit-command-arg "" --exit-command-arg --network-backend --exit-command-arg netavark --exit-command-arg --volumepath --exit-command-arg /var/lib/containers/storage/volumes --exit-command-arg --db-backend --exit-command-arg boltdb --exit-command-arg --transient-store=false --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg --storage-opt --exit-command-arg overlay.mountopt=nodev,metacopy=on --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg container --exit-command-arg cleanup --exit-command-arg 695b870aba3fd0de835ad37008a7375110d6da4ee21cff14b86e692378f2cc9c

Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: ++ cat /run_command
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + CMD=/usr/bin/neutron-dhcp-agent
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + ARGS=
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + sudo kolla_copy_cacerts
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + [[ ! -n '' ]]
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + . kolla_extend_start
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + echo 'Running command: '\''/usr/bin/neutron-dhcp-agent'\'''
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: Running command: '/usr/bin/neutron-dhcp-agent'
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + umask 0022
Apr 03 23:44:23 standalone.localdomain neutron_dhcp_agent[317396]: + exec /usr/bin/neutron-dhcp-agent

[ps -elf](https://logserver.rdoproject.org/49/349/7cb8b39e2b563d7c724a56b64d4a57c111ea621a/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/f9b09be/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/sos-reports/_all_nodes/sosreport-standalone.localdomain/sos_commands/process/ps_-elfL)
4 S 42435      97429   97415   97429  1    1  80   0 - 35784 ep_pol 23:00 ?        00:00:34 neutron-dhcp-agent (/usr/bin/python3 /usr/bin/neutron-dhcp-agent --config-file /usr/share/neutron/neutron-dist.conf --config-file /etc/neutron/neutron.conf --config-file /etc/neutron/dhcp_agent.ini --config-dir /etc/neutron/conf.d/common --config-dir /etc/neutron/conf.d/neutron-dhcp-agent --log-file=/var/log/neutron/dhcp-agent.log)

4 S root      317400  317398  317400  0    1  80   0 - 52736 ep_pol 23:44 ?        00:00:02 neutron-dhcp-agent (/usr/bin/python3 /usr/bin/neutron-dhcp-agent)

**Pull request merged**